### DISCUSSION TITLE
I need some help with one of my projects

### DISCUSSION THREAD

**Discussion by Kernix13** [EddieHubCommunity/support#3450] [Q&A]
Knowledge of Music Theory would be helpful, but not necessary. I am creating a Guitar Chord Name app where a user can enter fret #'s of an unknown chord. Clicking a Submit button writes the name of it onto the page, along with additional information. 

I have 8 tasks that I need to do and I'm stuck on #5. 

Overview: I am using an anonymous function in the event listener and calling a named function (`getNotes`) inside of it. In that function, I am:

1. Getting the input values (the fret #'s) (DONE)
1. Converting the fret #'s into chromatic notes (DONE)
1. Using `innerHTML` to add the chord tones onto the page (partially DONE)
1. Creating a 12-note array for each chord tone (DONE)
1. For each note, I need to find the interval for the other notes in relation to that note. I need to use the 12-note arrays from #4. (Why are there links attached to these points?)
1. Calculate the chord name based on #5
1. EXTRA: Output name(s) of chords that have the same notes, if applicable, requires #6
1. EXTRA: Output scale(s) that build that chord, requires #6

Right now I have a lot of variables in `script.js` but I am somehow going to have to pull in large JSON files. 

**PROBLEM**: How do I accomplish task #5? Right now I am doing everything inside of a `for` loop. I was thinking of using a `forEach` loop, which will work for the 1st item, but how do I cycle back for that item when I get to the 2nd item/note? And then cycle back for notes 1 and 2 when on note 3, etc?

Here is a [CodePen Example](https://codepen.io/jim-kernicky/pen/zYPvxda) and my [GitHub Repo](https://github.com/Kernix13/guitar-chord-names).

**Comment by hiteshvijayhv** [comment]
I don't know if I understand your project well but you can use a 2d array or a hashmap that has a pair of time of click and the note. And calculate each time you move from a origin to destination note.

**Comment by Kernix13** [comment]
I don't understand "hashmap" or "pair of time of click and the note". Is this in reference to #5, because I actually have that one solved but I have additional problems?
